src/js/shield.js

```diff
@@ -377,11 +377,23 @@ export function romanizeRef(ref) {
   return roman + ref.slice(number.toString().length);
 }
 
-export function generateShieldCtx(map, id) {
-  return generateSpriteCtx(map.style.imageManager.images, id);
+export function getDrawnShieldBounds(id) {
+  let routeDef = getRouteDef(id);
+  let shieldDef = getShieldDef(routeDef);
```

Comment:
Both of this function’s call sites already have `shieldDef` and `routeDef` in context that they can pass in. That would make it a little easier to unit test this function.

Reply:
Good call, fixed in bd85e51